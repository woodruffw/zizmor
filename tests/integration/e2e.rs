//! End-to-end snapshot integration tests.

use anyhow::Result;

use crate::common::{input_under_test, zizmor, OutputMode};

#[cfg_attr(not(feature = "gh-token-tests"), ignore)]
#[test]
fn gha_hazmat() -> Result<()> {
    // Stability test against with online retrieval but no online audits.
    // Ensures that we consistently collect the same files in the default
    // configuration.
    insta::assert_snapshot!(zizmor()
        .offline(false)
        .output(OutputMode::Both)
        .args(["--no-online-audits"])
        .input("woodruffw/gha-hazmat@42064a9533f401a493c3599e56f144918f8eacfd")
        .run()?);
    Ok(())
}

#[test]
fn menagerie() -> Result<()> {
    insta::assert_snapshot!(zizmor()
        .output(OutputMode::Both)
        .input(input_under_test("e2e-menagerie"))
        .run()?);

    Ok(())
}
