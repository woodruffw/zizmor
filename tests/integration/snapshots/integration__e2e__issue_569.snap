---
source: tests/integration/e2e.rs
expression: "zizmor().offline(false).output(OutputMode::Both).args([\"--no-online-audits\",\n\"--collect=workflows-only\"]).input(\"python/cpython@f963239ff1f986742d4c6bab2ab7b73f5a4047f6\").run()?"
snapshot_kind: text
---
 INFO zizmor: skipping impostor-commit: offline audits only requested
 INFO zizmor: skipping ref-confusion: offline audits only requested
 INFO zizmor: skipping known-vulnerable-actions: offline audits only requested
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/add-issue-header.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/build.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/documentation-links.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/jit.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/lint.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/mypy.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/new-bugs-announce-notifier.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/project-updater.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/require-pr-label.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-context.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-docs.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-macos.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-tsan.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-ubuntu.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-wasi.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-windows-msi.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/reusable-windows.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/stale.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/tail-call.yml
 INFO audit: zizmor: ðŸŒˆ completed .github/workflows/verify-ensurepip-wheels.yml
error[dangerous-triggers]: use of fundamentally insecure workflow trigger
  --> .github/workflows/documentation-links.yml:5:1
   |
 5 | / on:
 6 | |   pull_request_target:
...  |
10 | |     - 'Doc/**'
11 | |     - '.github/workflows/doc.yml'
   | |_________________________________^ pull_request_target is almost always used insecurely
   |
   = note: audit confidence â†’ Medium

104 findings (103 suppressed): 0 unknown, 0 informational, 0 low, 0 medium, 1 high
